### ISSUE TITLE
Setting logLevel on specific host fails

### ISSUE THREAD

**Issue by heschlie** [projectcalico/calico#1033]
When setting the logLevel via calicoctl for a specific Node it fails.

## Expected Behavior

`calicoctl config set logLevel warning --node=$NODE` should annotate the Node with the new log level.

## Current Behavior

An error is thrown:
```
./calicoctl config set logLevel warning --node=172.18.18.102
Error executing command: operation Apply is not supported on HostConfig(node=172.18.18.102,name=LogSeverityScreen)
```

## Possible Solution

Allow Apply on HostConfig?

## Steps to Reproduce (for bugs)

1. Fire up Vagrant cluster via 2.5 docs
2. Deploy KDD Calico-v2.5-rc2
3. Issue `calicoctl config set logLevel --node=$NODE`

## Your Environment

* Calico version: v2.5-rc2
* Orchestrator version (e.g. kubernetes, mesos, rkt): k8s
* Operating System and version: CoreOS
* Link to your project (optional):

**Comment by heschlie** [comment]
Debug output:

```
/calicoctl -l debug config set logLevel warning --node="172.18.18.102"         
INFO[0000] Log level set to debug                       
INFO[0000] Config file: /etc/calico/calicoctl.cfg cannot be read - reading config from environment 
INFO[0000] Loading config from environment              
INFO[0000] Loaded client config: api.CalicoAPIConfigSpec{DatastoreType:"kubernetes", EtcdConfig:api.EtcdConfig{EtcdScheme:"http", EtcdAuthority:"127.0.0.1:2379", EtcdEndpoints:"", EtcdUsername:"", EtcdPassword:"", EtcdKeyFile:"", EtcdCertFile:"", EtcdCACertFile:""}, KubeConfig:api.KubeConfig{Kubeconfig:"", K8sAPIEndpoint:"http://127.0.0.1:8080", K8sKeyFile:"", K8sCertFile:"", K8sCAFile:"", K8sAPIToken:"", K8sInsecureSkipTLSVerify:false, K8sDisableNodePoll:false}} 
DEBU[0000] Using datastore type 'kubernetes'            
DEBU[0000] Building client for config: &{Kubeconfig: K8sAPIEndpoint:http://127.0.0.1:8080 K8sKeyFile: K8sCertFile: K8sCAFile: K8sAPIToken: K8sInsecureSkipTLSVerify:false K8sDisableNodePoll:false} 
DEBU[0000] Config overrides: &{AuthInfo:{LocationOfOrigin: ClientCertificate: ClientCertificateData:[] ClientKey: ClientKeyData:[] Token: TokenFile: Impersonate: Username: Password: AuthProvider:<nil> Extensions:map[]} ClusterDefaults:{LocationOfOrigin: Server: APIVersion: InsecureSkipTLSVerify:false CertificateAuthority: CertificateAuthorityData:[] Extensions:map[]} ClusterInfo:{LocationOfOrigin: Server:http://127.0.0.1:8080 APIVersion: InsecureSkipTLSVerify:false CertificateAuthority: CertificateAuthorityData:[] Extensions:map[]} Context:{LocationOfOrigin: Cluster: AuthInfo: Namespace: Extensions:map[]} CurrentContext: Timeout:} 
DEBU[0000] Created k8s clientSet: &{DiscoveryClient:0xc4203e8ac0 CoreV1Client:0xc420479e30 AppsV1beta1Client:0xc420479fd0 AuthenticationV1Client:0xc4202a40f0 AuthenticationV1beta1Client:0xc4202a42b0 AuthorizationV1Client:0xc4202a43d0 AuthorizationV1beta1Client:0xc4202a44d0 AutoscalingV1Client:0xc4202a45d0 AutoscalingV2alpha1Client:0xc4202a4730 BatchV1Client:0xc4202a4850 BatchV2alpha1Client:0xc4202a4980 CertificatesV1beta1Client:0xc4202a4a80 ExtensionsV1beta1Client:0xc4202a4bb0 PolicyV1beta1Client:0xc4202a4d50 RbacV1beta1Client:0xc4202a4fb0 RbacV1alpha1Client:0xc4202a50b0 SettingsV1alpha1Client:0xc4202a5220 StorageV1beta1Client:0xc4202a5370 StorageV1Client:0xc4202a54b0} 
DEBU[0000] Performing 'Apply' for &{Key:HostConfig(node=172.18.18.102,name=LogSeverityScreen) Value:warning Revision:<nil> TTL:0s} 
WARN[0000] Attempt to 'Apply' using kubernetes backend is not supported. 
DEBU[0000] Performing 'Apply' for &{Key:HostBGPConfig(node=172.18.18.102; name=loglevel) Value:none Revision:<nil> TTL:0s} 
DEBU[0000] Apply per-Node resource                       Key=HostBGPConfig(node=172.18.18.102; name=loglevel) Namespace=config.bgp.projectcalico.org Resource=NodeBGPConfig Value=none
DEBU[0000] Extract node and resource info from Key       Key=HostBGPConfig(node=172.18.18.102; name=loglevel) Namespace=config.bgp.projectcalico.org Resource=NodeBGPConfig
DEBU[0000] Updating value in annotation                  Action=Apply Namespace=config.bgp.projectcalico.org Resource=NodeBGPConfig Value=none
Error executing command: operation Apply is not supported on HostConfig(node=172.18.18.102,name=LogSeverityScreen)
```

**Comment by heschlie** [comment]
Apparently this isn't supported by KDD yet, closing

**Comment by caseydavenport** [comment]
This should be fixed in the latest v3.0 alpha. I'll re-test before closing.

**Comment by caseydavenport** [comment]
Retested and it works!
